### ISSUE TITLE
[MNG-5346] update maven-plugin-plugin:descriptor default binding from generate-resources phase to process-classes

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#6917]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** opened **[MNG-5346](https://issues.apache.org/jira/browse/MNG-5346?redirect=false)** and commented

with Java annotations support added in Maven Plugin Tools 3.0, descriptor cannot be generated before compilation

actually, to use annotations, users need to add extra configuration to avoid failure and to bind the goal to proper phase:

```xml
<configuration>
  <skipErrorNoDescriptorsFound>true</skipErrorNoDescriptorsFound>
</configuration>
<executions>
  <execution>
    <id>mojo-descriptor</id>
    <goals>
      <goal>descriptor</goal>
    </goals>
  </execution>
```

changing the default lifecycle binding will enable removal of this extra configuration

notice that removing the configuration from pom will require to check newer Maven version is used to build the plugin

---

**Affects:** 3.2.1

**Attachments:**
- [plugin-annotation-sample.zip](https://issues.apache.org/jira/secure/attachment/12713335/plugin-annotation-sample.zip) (_1.71 kB_)

**Issue Links:**
- [MNG-6169](https://issues.apache.org/jira/browse/MNG-6169) Packaging plugin bindings version updates

- [MPLUGIN-238](https://issues.apache.org/jira/browse/MPLUGIN-238) HelpMojo phases fails when using java-annotations

7 votes, 16 watchers

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

The current solution to get it working without the above skipErrorNoDescriptorsFound configuration:

```
<plugins>
  <plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-plugin-plugin</artifactId>
    <configuration>
      <goalPrefix>configurator</goalPrefix>
    </configuration>
    <executions>
      <execution>
        <id>default-descriptor</id>
        <goals>
          <goal>descriptor</goal>
        </goals>
        <phase>process-classes</phase>
      </execution>
      <execution>
        <id>help-descriptor</id>
        <goals>
          <goal>helpmojo</goal>
        </goals>
        <phase>process-classes</phase>
      </execution>
    </executions>
  </plugin>
</plugins>
```

**Comment by jira-importer** [comment]
**[Jason van Zyl](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jvanzyl)** commented

Please refer to https://cwiki.apache.org/confluence/display/MAVEN/The+Great+JIRA+Cleanup+of+2014

**Comment by jira-importer** [comment]
**[Jason van Zyl](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jvanzyl)** commented

Please refer to https://cwiki.apache.org/confluence/display/MAVEN/The+Great+JIRA+Cleanup+of+2014 if you're wondering why this issue was closed out.

**Comment by jira-importer** [comment]
**[Jan Sievers](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jan.sievers)** commented

the problem is still there, see sample project attached.

Also, the docs
http://maven.apache.org/plugin-tools/maven-plugin-annotations/

do not mention this bug.
Anyone following the instructions in the docs will fail and end up spending a lot of time googling why it does not work.

I'd ike to re-open this issue as advertised on
https://cwiki.apache.org/confluence/display/MAVEN/The+Great+JIRA+Cleanup+of+2014

but it seems I don't have permissions. Please re-open or provide steps where I can open a bug instead.

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

it's not exactly a bug, and as explained in MPLUGIN-238, it's documented in http://maven.apache.org/plugin-tools/maven-plugin-plugin/examples/using-annotations.html
but you're right, http://maven.apache.org/plugin-tools/maven-plugin-annotations/ should link to that page: that's a good idea for MPLUGIN-238
any other good idea in MPLUGIN-238 is welcome

now, reopening this issue in Maven Core, to avoid the problem is IMHO a good idea, since it would be better

**Comment by jira-importer** [comment]
**[Jason van Zyl](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jvanzyl)** commented

Ok, I will make this change for the 3.2.2 release.

**Comment by jira-importer** [comment]
**[Jason van Zyl](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jvanzyl)** commented

Fixed on fc5cd371095913115f6cda6607630ba452a1b8bc
